File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public static void reverse(int[] nums, int k, int i, int j){
        while(i<j){
            int temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
            i++;
            j--;
        }
    }


    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k=k%n;
        reverse(nums, k, 0, nums.length-1);
        reverse(nums, k, 0, k-1);
        reverse(nums, k, k, nums.length-1);
        
    }
}

File: 0189-rotate-array/0189-rotate-array.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        list = [0]*n
        for i, num in enumerate(nums):
            list[(i+k)%n] = num
        for i, num in enumerate(list):
            nums[i] = num
            
        